File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/ClockModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable;

import org.threeten.bp.Clock;

import dagger.Module;
import dagger.Provides;
import dagger.Reusable;

@Module
public abstract class ClockModule {

	// Published
	@Provides
	@Reusable
	static Clock provideClock() {
		return Clock.systemUTC();
	}

}

File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/ClockModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable

import dagger.Module
import dagger.Provides
import dagger.Reusable
import org.threeten.bp.Clock

@Module
class ClockModule {

    // Published
    @Provides
    @Reusable
    internal fun provideClock(): Clock {
        return Clock.systemUTC()
    }

}
